src/main/java/net/snowflake/client/core/JsonSqlOutput.java

```diff
@@ -0,0 +1,390 @@
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "fixed", 0, 38, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeInt(int input) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, input);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "fixed", 0, 38, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeLong(long value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "fixed", 0, 38, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeFloat(float value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeDouble(double value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeBigDecimal(BigDecimal value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, value);
+          schema
+              .getFields()
+              .add(buildSchemaWithScaleAndPrecision(fieldName, "real", 0, 0, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeBytes(byte[] value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(fieldName, new SFBinary(value).toHex());
+          schema
+              .getFields()
+              .add(FieldSchemaCreator.buildSchemaForBytesType(fieldName, maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeDate(Date value) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          json.put(
+              fieldName,
+              ResultUtil.getDateAsString(value, getDateTimeFormat("DATE_OUTPUT_FORMAT")));
+          schema
+              .getFields()
+              .add(FieldSchemaCreator.buildSchemaTypeAndNameOnly(fieldName, "date", maybeColumn));
+        }));
+  }
+
+  @Override
+  public void writeTime(Time x) throws SQLException {
+    withNextValue(
+        ((json, fieldName, maybeColumn) -> {
+          final long MS_IN_DAY = 86400 * 1000;
+          long msSinceEpoch = x.getTime();
+          long msSinceMidnight = (msSinceEpoch % MS_IN_DAY + MS_IN_DAY) % MS_IN_DAY;
```

Comment:
Why can't we just use `msSinceEpoch % MS_IN_DAY`?

Reply:
It was get from existed implementation. I extracted this part and reuse it.